docs/source/notebooks/Gedik_qudit.ipynb

```diff
@@ -571,40 +559,28 @@
     }
    ],
    "source": [
-    "# Run the algorithm on the QPU for all the permutations.\n",
-    "res_0 = algo_qpu(0, qft_unit, inv_qft_unit)\n",
-    "res_1 = algo_qpu(1, qft_unit, inv_qft_unit)\n",
-    "res_2 = algo_qpu(2, qft_unit, inv_qft_unit)\n",
-    "res_3 = algo_qpu(3, qft_unit, inv_qft_unit)\n",
-    "res_4 = algo_qpu(4, qft_unit, inv_qft_unit)\n",
-    "res_5 = algo_qpu(5, qft_unit, inv_qft_unit)\n",
-    "res_6 = algo_qpu(6, qft_unit, inv_qft_unit)\n",
-    "res_7 = algo_qpu(7, qft_unit, inv_qft_unit)"
+    "# Run the algorithm on the QPU for all the permutations\n",
+    "jobs = []\n",
+    "for i in range(8):\n",
+    "    jobs.append(algo_qpu(i, qft_unit, inv_qft_unit))"
    ]
   },
   {
    "cell_type": "code",
-   "execution_count": 17,
-   "id": "f34c4d14",
+   "execution_count": 41,
+   "id": "9df02c29",
    "metadata": {},
    "outputs": [],
    "source": [
-    "# Get the results from the QPU implementation for each permutation.\n",
+    "# Get the results once they are available\n",
     "data_qpu = []\n",
-    "\n",
-    "data_qpu.append(res_0.get_results()['results'])\n",
-    "data_qpu.append(res_1.get_results()['results'])\n",
-    "data_qpu.append(res_2.get_results()['results'])\n",
-    "data_qpu.append(res_3.get_results()['results'])\n",
-    "data_qpu.append(res_4.get_results()['results'])\n",
-    "data_qpu.append(res_5.get_results()['results'])\n",
-    "data_qpu.append(res_6.get_results()['results'])\n",
-    "data_qpu.append(res_7.get_results()['results'])"
+    "for i in range(8):\n",
+    "    data_qpu.append(jobs[i].get_results()['results'])"
```

Comment:
Jobs are ran asynchronously so they might run for long and the notebook would go on. Is there anything checking that the jobs have finished before the `get_results()` call ? If not, I guess the notebook is not re-runnable by a user, as it would require an arbitrary waiting time between this cell and the previous one.